### ISSUE TITLE
Allow custom "Stringifier"

### ISSUE THREAD

**Issue by henriquemoody** [Respect/Validation#1227]
# Problem
When we create the exception message, we use the `stringify` function (from [Respect\Stringifier]) to convert the input [1] all the properties that come from the rule [2] to a string.

There are some cases, described in #1048 and #1157 in which a user would like to have a different way to convert those values to string.

# Proposed solution

Pass [Stringifier] to the [Factory] and then to the [ValidationException], so that users can customize how to stringify values.

[1]: https://github.com/Respect/Validation/blob/718bacad046b5f636c33ab0f9d37bca4dbacb149/library/Exceptions/ValidationException.php#L155
[2]: https://github.com/Respect/Validation/blob/718bacad046b5f636c33ab0f9d37bca4dbacb149/library/Exceptions/ValidationException.php#L187
[Respect\Stringifier]: https://github.com/Respect/Stringifier
[Stringifier]: https://github.com/Respect/Stringifier/blob/master/src/Stringifier.php
[Factory]: https://github.com/Respect/Validation/blob/718bacad046b5f636c33ab0f9d37bca4dbacb149/library/Factory.php
[ValidationException]: https://github.com/Respect/Validation/blob/718bacad046b5f636c33ab0f9d37bca4dbacb149/library/Exceptions/ValidationException.php

**Comment by mazentouati** [comment]
Hey @henriquemoody could you please provide an example how the user would use the API to customize the Stringifier based on the proposed solution ?

**Comment by henriquemoody** [comment]
Basically:

```php
Factory::setDefaultInstance(
    new Factory(
        ['My\Validation\Rules'],
        ['My\Validation\Exceptions'],
        $translator,
        $stringifier,
    )
);
```

We could maybe create a named constructor:


```php
Factory::setDefaultInstance(Factory::fromStringifier($stringifier));
```

***
Later on would be nice to improve the documentation from https://github.com/Respect/Stringifier

**Comment by henriquemoody** [comment]
Fixed by #1234
